zerver/webhooks/semaphore2/view.py

```diff
@@ -0,0 +1,184 @@
+        commit_message = payload["revision"]["commit_message"]
+        commit_url = get_commit_url(commit_id, repository_url, repository_slug)
+        content = PUSH_TEMPLATE.format(
+            email = author_email,
+            branch_name = branch_name,
+            organization_name = organization_name,
+            project_name = project_name,
+            repository_url = repository_url,
+            commit_hash = commit_id[:7],
+            commit_message = commit_message,
+            commit_url = commit_url,
+            pipeline_name = pipeline_name,
+            pipeline_result = pipeline_result,
+            blocks = blocks,
+        )
+
+    elif event == "pull_request":
+        content = PULL_REQUEST_TEMPLATE.format(
+            email = author_email,
+            branch_name = branch_name,
+            organization_name = organization_name,
+            project_name = project_name,
+            repository_url = repository_url,
+            pull_request_title = pull_request_title,
+            pull_request_url = pull_request_url,
+            pipeline_name = pipeline_name,
+            pipeline_result = pipeline_result,
+            blocks = blocks,
+        )
+
+    else:  # should never get here
+        content = DEFAULT_TEMPLATE.format(
+            email = author_email,
+            branch_name = branch_name,
+            organization_name = organization_name,
+            project_name = project_name,
+            repository_url = repository_url,
+            pipeline_name = pipeline_name,
+            pipeline_result = pipeline_result,
+            blocks = blocks,
+        )
+
+    subject = TOPIC_TEMPLATE.format(
+        project=project_name,
+        branch=branch_name
+    )
+
+    check_send_webhook_message(request, user_profile, subject, content)
+    return json_success()
+
+#Helper Functions
+def get_blocks_data(blocks: List[Dict[str, Any]]) -> str:
+    """
+        Returns the markdown presentation of data related to blocks.
+    """
+    BLOCKS_DATA = """"""
+    BLOCK_BODY = """
+**1. {name}**:
+  - State: {state}
+  - Result: {result}
+""".strip()
+    for block in blocks:
+        name = block["name"].title()
+        state = block["state"]
+        result = block["result"]
+        content = BLOCK_BODY.format(name=name, state=state, result=result)
+        BLOCKS_DATA += content
+
+    return BLOCKS_DATA
+
+# Funtions to generate url to a commit/Pull Request. This builds an url
+# as per the general pattern of github urls.
+def get_commit_url(commit_id: str, repo_url: str, repo_slug: str) -> str:
+    """
+        Returns the url of a particular commit using the commit_id and
+        repository url.
+    """
+    repository_hosting_services = {
+        "https://github.com": repo_url + '/commit/' + commit_id
+    }
+    # New hosting services must be added when semaphore extends it support.
+
+    repo_slug = '/' + repo_slug
+    repository_hosting_service_url = repo_url.replace(repo_slug, '')
```

Comment:
I would parse the URL and get the domain name/host, instead of this string manipulation.

Reply:
I'll use `urllib.parse` and get the domain name, that would be a lot better. Thanks!